### PR TITLE
subapp paths now return the correct values when mounted on array of paths. fixes #389

### LINKED ISSUE #389: Mounting a subapp on array of paths
**Describe the bug**

The introduction of array of paths as argument #379 has caused an issue with subapp mounting. When a subapp is mounted on an array of paths, logically speaking it has no true mount path. Instead, as of the current release, it will output a default `/`.

**To Reproduce**

Run this code

```javascript
import { App } from '@tinyhttp/app';

const app = new App()
const subapp = new App()
const subsubapp = new App()

app.use(['/t1', '/t2'], subapp)
subapp.use(['/t3', '/t4'], subsubapp)

console.log(subsubapp.path())

app.listen(3000, () => {
    console.log('listening!')
})
```
Current output is `//`

**Expected behavior**

I am unsure. Maybe it should throw an error? Opinions are welcome

**Versions**

- `node`: 16.20.0
- `@tinyhttp/app`: 2.0.33

**Additional context**

None

### PR DESCRIPTION
**Pull request by aarontravass**
`app.path()` would default to `//` when mounted for array of paths. This fixes it and uses express style mounting.
Added tests for coverage as well

fixes #389

### CODE REVIEW

**Comment by aarontravass** [comment]
let me increase cov

**Comment by aarontravass** [comment]
gh actions does not symlinks I guess :(

**Comment by aarontravass** [comment]
@talentlessguy is there any other way to get the last branch? https://github.com/tinyhttp/tinyhttp/blob/6b3944276b022702f4699a4e925b823e1b9b8ce5/packages/app/src/app.ts#L172

I can only think of symlinks

File: packages/jsonp/src/index.ts [hunk 1]
```diff
@@ -25,8 +25,6 @@ function stringify(
           return '\\u003e'
         case 0x26:
           return '\\u0026'
-        default:
```
Comment (talentlessguy) [inline packages/jsonp/src/index.ts:28]: why remove this?

File: tests/core/app.test.ts [hunk 2]
```diff
@@ -934,6 +976,27 @@ describe('Template engines', () => {
 
     await fetch('/').expectBody('Hello World')
   })
+  it.skip('can render without options passed', async () => {
+    // github actions does not support symlink
```
Comment (talentlessguy) [inline tests/core/app.test.ts:980]: this test is unrelated to the PR

**Comment by talentlessguy** [comment]
there's a lot of changes unrelated to the PR itself, let's move it to a separate one

**Comment by aarontravass** [comment]
> there's a lot of changes unrelated to the PR itself, let's move it to a separate one

yeah. im gonna close this open a new one
